assets/src/components/parts/janus-input-number/InputNumber.tsx

```diff
@@ -204,50 +236,40 @@ const InputNumber: React.FC<PartComponentProps<InputNumberModel>> = (props) => {
     if (width !== undefined) {
       styleChanges.width = { value: width as number };
     }
-    props.onResize({ id: `${id}`, settings: styleChanges });
-  }, [width]);
+    onResize({ id: `${id}`, settings: styleChanges });
+  }, [id, width, onResize]);
+
+  const saveInputText = useCallback(
+    (val: number) => {
+      onSave({
+        id: `${id}`,
+        responses: [
+          {
+            key: 'value',
+            type: CapiVariableTypes.NUMBER,
+            value: val,
+          },
+        ],
+      });
+    },
+    [id, onSave],
+  );
 
   const debouncetime = 300;
+  // eslint-disable-next-line react-hooks/exhaustive-deps
   const debounceSave = useCallback(
     debounce((val) => {
       saveInputText(val);
     }, debouncetime),
-    [],
+    [saveInputText],
   );
 
-  const saveInputText = (val: number) => {
-    props.onSave({
-      id: `${id}`,
-      responses: [
-        {
-          key: 'value',
-          type: CapiVariableTypes.NUMBER,
-          value: val,
-        },
-      ],
-    });
-  };
-
   const handleOnChange: ReactEventHandler<HTMLInputElement> = (event) => {
-    const val = (event.target as HTMLInputElement).value;
+    const val = sanitizeValue((event.target as HTMLInputElement).value);
```

Comment:
This is the main point of this PR.
Before, the user would type, we would call setInputNumberValue
Later, an effect would see that value changed and call debounceSave 
Eventually that would trigger and the value would get passed up.

Unfortunately, we call setInputNumberValue for several other reasons that are not user-input. Each of those would cause a value change, effect trigger, and a debounceSave

We would also always get a debounceSave on the initial rendering since the value was new.

Now, we directly call debounceSave from the change handler, so only user input trigers it.